src/gl.rs

```diff
@@ -67,6 +67,16 @@ pub fn read_buffer(mode: GLenum) {
     }
 }
 
+pub fn read_pixels_into_buffer(x: GLint, y: GLint, width: GLsizei, height: GLsizei,
```

Comment:
How about we make the dst_buffer param a mutable slice, and then abstract the code in read_pixels() that calculates the length into a function? Then we could call that function here to assert that the length of the slice matches the expected length.